contexts/SubscriptionContext/tests/Integration/UseCases/AddSubscription/AddSubscriptionUseCaseTest.php

```diff
@@ -139,4 +141,19 @@ public function testGivenDataThatNeedsToBeModerated_requestNotBeMailed() {
 		$useCase->addSubscription( $request );
 	}
 
+	public function testNewMailAddressFromSubscription(): void {
+		$useCase = new AddSubscriptionUseCase( $this->repo, $this->validator, $this->mailer );
+		$class = new ReflectionClass( AddSubscriptionUseCase::class );
```

Comment:
> not using reflection, claiming that only public API needs testing

I think that is a good rule by and large... there might be exceptions, esp in legacy codebases... though I can't think of this being called for in code that was TDD'd or well designed code in general. If you have another opinion I'd love to hear your rationale.